caikit_nlp/modules/token_classification/transformer_sentence_classification.py

```diff
@@ -0,0 +1,218 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module composes sentence splitting and sequence classification.
+At this time this module is only designed for inference"""
+
+# Standard
+from typing import Dict, List, Optional
+import os
+
+# First Party
+from caikit.core.modules import (
+    ModuleBase,
+    ModuleConfig,
+    ModuleLoader,
+    ModuleSaver,
+    module,
+)
+from caikit.core.toolkit import error_handler
+import alog
+
+# Local
+from ...data_model import TokenClassification, TokenClassificationResult
+from ..sentence_split.base import SentenceSplitBase
+from ..text_classification import SequenceClassification
+from .token_classification_task import TokenClassificationTask
+
+log = alog.use_channel("SEQ_TSFMR")
+error = error_handler.get(log)
+
+
+@module(
+    id="42a7d920-8b7e-4e1f-81fb-8ab851a80c99",
+    name="Transformer sentence-level classification",
+    version="0.1.0",
+    task=TokenClassificationTask,
+)
+class TransformerSentenceClassification(ModuleBase):
+
+    ################################ Constructor #################################################
+
+    def __init__(
+        self,
+        lang: str,
+        sentence_splitter: SentenceSplitBase,
+        sequence_classifier: SequenceClassification,
+        default_threshold: float,
+        labels_to_output: List[str] = None,
+        labels_mapping: Dict[str, str] = None,
+    ):
+        """Construct a transformer sentence classification object
+        from a sentence splitter and sequence classifier
+
+        Args:
+            lang: str
+                2 letter language code
+            sentence_splitter: SentenceSplitBase
+                Sentence splitter that returns List[Span]
+            sequence_classifier: SequenceClassification
+                Sequence tokenizer and classification model
```

Comment:
This string seems a bit wrong?

Reply:
Well we're assuming it's the huggingface pattern with the tokenizer and classifier bundled together, but I can just clarify to say it is `SequenceClassification` model